sw/device/lib/crypto/impl/aes_gcm.c

```diff
@@ -66,6 +66,34 @@ static inline void gcm_context_restore(otcrypto_aes_gcm_context_t *api_ctx,
                   kAesGcmContextNumWords);
 }
 
+/**
+ * Remask the key if it is not sideloaded.
+ *
+ * Generate a fresh mask and apply it to the current key.
+ *
+ * @param[in,out] internal_ctx Internal context object.
+ * @return Result of the operation.
+ */
+status_t gcm_remask_key(aes_gcm_context_t *internal_ctx) {
+  if (launder32(internal_ctx->key.sideload) == kHardenedBoolFalse) {
+    HARDENED_CHECK_EQ(internal_ctx->key.sideload, kHardenedBoolFalse);
+
+    // Generate a fresh mask the size of one share.
+    uint32_t mask[internal_ctx->key.key_len];
+    hardened_memshred(mask, internal_ctx->key.key_len);
+
+    // XOR each share with the mask.
+    HARDENED_TRY(hardened_xor((uint32_t *)internal_ctx->key.key_shares[0], mask,
+                              internal_ctx->key.key_len));
+    HARDENED_TRY(hardened_xor((uint32_t *)internal_ctx->key.key_shares[1], mask,
+                              internal_ctx->key.key_len));
+  } else {
+    HARDENED_CHECK_EQ(internal_ctx->key.sideload, kHardenedBoolTrue);
```

Comment:
Shall we return error upon calling remask with a sideload key to notify user? 

sideloaded keys are always remasked on every power cycle iiuc